### PR TITLE
[GEN-615] Enabling iOS Push Notification & making Record the default landing tab

### PR DESCRIPTION
**Pull request by VGmt**

### CODE REVIEW

File: trk/src/Navigation/AppNavigator.js [hunk 1]
```diff
@@ -431,16 +435,25 @@ function AppTabs() {
...
   if (enabled) {
     console.log('Authorization status:', authStatus);
+    // Now fetch the FCM token
+    const fcmToken = await messaging().getToken();
+    if (fcmToken) {
+      console.log('FCM Token:', fcmToken);
+      // Perform any additional setup with the FCM token, like sending it to your server
+    } else {
+      console.log('Failed to fetch FCM token');
+    }
```
Comment (VGmt) [inline trk/src/Navigation/AppNavigator.js:451]: logging fcm token when `enabled`.

File: trk/src/Navigation/AppNavigator.js [hunk 2]
```diff
@@ -343,7 +347,7 @@ function AppTabs() {
   const { role } = useContext(AuthContext);
 
   return (
-    <Tab.Navigator>
+    <Tab.Navigator initialRouteName="Record">
```
Comment (VGmt) [inline trk/src/Navigation/AppNavigator.js:350]: make Record default tab

File: trk/ios/trk/AppDelegate.mm [hunk 3]
```diff
@@ -2,23 +2,37 @@
...
+    [center requestAuthorizationWithOptions:authOptions
+                          completionHandler:^(BOOL granted, NSError * _Nullable error) {
+                              // Handle granting of permissions
+                          }];
+  }
+  [application registerForRemoteNotifications];
+
+  // Set the messaging delegate
+  [FIRMessaging messaging].delegate = self;
+
```
Comment (VGmt) [inline trk/ios/trk/AppDelegate.mm:35]: enable iOS push notif compatibility

File: trk/ios/trk/AppDelegate.mm [hunk 4]
```diff
@@ -31,4 +45,42 @@ - (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
...
+- (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
+  // With Firebase SDK 10.4.0 and later, direct setting of the APNs token here is necessary.
+  [FIRMessaging messaging].APNSToken = deviceToken;
+}
+
+- (void)application:(UIApplication *)application didFailToRegisterForRemoteNotificationsWithError:(NSError *)error {
+  NSLog(@"Failed to register for remote notifications: %@", error);
+}
+
```
Comment (VGmt) [inline trk/ios/trk/AppDelegate.mm:85]: same here – enabling iOS push notif

File: trk/ios/trk/AppDelegate.mm [hunk 5]
```diff
@@ -2,23 +2,37 @@
 #import <Firebase.h>
 #import <React/RCTBundleURLProvider.h>
 #import <React/RCTLinkingManager.h>
+#import <UserNotifications/UserNotifications.h>
+#import <FirebaseMessaging/FirebaseMessaging.h>
+#import <React/RCTBridge.h>
```
Comment (VGmt) [inline trk/ios/trk/AppDelegate.mm:7]: iOS push notif imports

**Review by abhipi — APPROVED** [review]
Looks good!

**Pull request merged**